website/content/en/docs/kubebuilder/references/client.md

```diff
@@ -0,0 +1,574 @@
+
+#### Get
+
+```Go
+// Get retrieves an API object for a given object key from the Kubernetes cluster
+// and stores it in obj.
+func (c Client) Get(ctx context.Context, key ObjectKey, obj runtime.Object) error
+```
+**Note**: An `ObjectKey` is simply a `client` package alias for [`types.NamespacedName`][doc-types-nsname].
+
+Example:
+```Go
+import (
+	"context"
+	"github.com/example-org/app-operator/pkg/apis/cache/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	app := &v1alpha1.App{}
+	ctx := context.TODO()
+	err := r.Get(ctx, request.NamespacedName, app)
+
+	...
+}
+```
+
+#### List
+
+```Go
+// List retrieves a list of objects for a given namespace and list options
+// and stores the list in obj.
+func (c Client) List(ctx context.Context, list runtime.Object, opts ...client.ListOption) error
+```
+
+A `client.ListOption` is an interface that sets [`client.ListOptions`][list-options] fields. A `client.ListOption` is created by using one of the provided implementations: [`MatchingLabels`][matching-labels], [`MatchingFields`][matching-fields], [`InNamespace`][in-namespace].
+
+Example:
+
+```Go
+import (
+	"context"
+	"fmt"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	// Return all pods in the request namespace with a label of `app=<name>`
+	// and phase `Running`.
+	podList := &v1.PodList{}
+	opts := []client.ListOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"app": request.NamespacedName.Name},
+		client.MatchingFields{"status.phase": "Running"},
+	}
+	ctx := context.TODO()
+	err := r.List(ctx, podList, opts...)
+
+	...
+}
+```
+
+[list-options]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#ListOptions
+[matching-labels]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#MatchingLabels
+[matching-fields]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#MatchingFields
+[in-namespace]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#InNamespace
+
+#### Create
+
+```Go
+// Create saves the object obj in the Kubernetes cluster.
+// Returns an error
+func (c Client) Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error
+```
+
+A `client.CreateOption` is an interface that sets [`client.CreateOptions`][create-options] fields. A `client.CreateOption` is created by using one of the provided implementations: [`DryRunAll`][dry-run-all], [`ForceOwnership`][force-ownership]. Generally these options are not needed.
+
+Example:
+
+```Go
+import (
+	"context"
+	"k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Update for new scaffold. `request reconcile.Request`==>`req ctrl.Request`
Update the imports as well `"sigs.k8s.io/controller-runtime/pkg/reconcile"`==>`ctrl "sigs.k8s.io/controller-runtime"`